platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ExportMigrationConfigurationAdminContext.java

```diff
@@ -0,0 +1,208 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.configuration.admin;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.Validate;
+import org.apache.felix.fileinstall.internal.DirectoryWatcher;
+import org.codice.ddf.configuration.persistence.PersistenceStrategy;
+import org.codice.ddf.migration.ExportMigrationContext;
+import org.codice.ddf.migration.ExportMigrationContextProxy;
+import org.codice.ddf.migration.ExportMigrationEntry;
+import org.codice.ddf.migration.MigrationEntry;
+import org.codice.ddf.migration.MigrationWarning;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class extends on the {@link ExportMigrationContext} interface to pre-create entries for
+ * all configuration objects.
+ */
+public class ExportMigrationConfigurationAdminContext extends ExportMigrationContextProxy {
+    private static final Logger LOGGER = LoggerFactory.getLogger(
+            ExportMigrationConfigurationAdminContext.class);
+
+    private final ConfigurationAdminMigratable admin;
+
+    private final Set<String> warnedExtensions = new HashSet<>(8);
+
+    private final Map<Path, ExportMigrationEntry> entries;
+
+    public ExportMigrationConfigurationAdminContext(ExportMigrationContext context,
+            ConfigurationAdminMigratable admin, Configuration[] configs) {
+        super(context);
+        Validate.notNull(admin, "invalid null configuration admin migratable");
+        Validate.notNull(configs, "invalid null configurations");
+        this.admin = admin;
+        this.entries = Stream.of(configs)
+                .filter(this::isValid)
+                .map(this::getEntry)
+                .collect(Collectors.toMap(MigrationEntry::getPath, Function.identity()));
+    }
+
+    /**
+     * This method is designed to circumvent a bug in Felix where it has been seen that a config object
+     * that is meant to represent a managed service factory is not properly created. Instead of being
+     * create using the createFactoryConfiguration(), it is obtained using getConfiguration(). After
```

Comment:
✏️  Typo: _created_ (missing _d_)

Reply:
I was trying to reduce the size of this PR 😛